cluster-autoscaler/processors/binpacking/time_limiter.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package binpacking
+
+import (
+	"time"
+
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	"k8s.io/autoscaler/cluster-autoscaler/expander"
+	"k8s.io/klog/v2"
+)
+
+// TimeLimiter expands binpackingLimiter to limit the total time spends on binpacking
+type TimeLimiter struct {
+	limiter               BinpackingLimiter
```

Comment:
Why does it wrap another limiter? For time-based only this is not needed, for multiple limiting criteria I'd suggest a separate limiter combining other limiters.

Reply:
Added a separate limiter for combining multiple limiters.